text/0000-config-transaction-library.md

```diff
@@ -0,0 +1,80 @@
+nav_order: 3
+---
+
+- Feature Name: Create a standalone config transaction library that generates channel config creation and update envelopes
+- Start Date: 2020-03-20
+- RFC PR: (leave this empty)
+- Fabric Component: fabric, fabric-config
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This feature will implement a consumable standalone library that supports generating configuration envelopes for 
+operations including application and system channel creation, channel configuration updates, and attaching 
+endorsing signatures to generated envelopes.
+
+# Motivation
+[motivation]: #motivation
+
+The motivation for this work is to provide a consistent API that is actually intended for consumption by external
+users for any operations involving channel config transactions. Currently `configtxgen` is available and frequently
+misused as a tool for these operations, but it was never intended for production usage. Much of the existing process for
+generating configuration updates is both tedious and error prone involving decoding a config transaction to JSON
+with the protolator tool and manually modifying fields prior to re-encoding back to proto and submitting for update.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Work on this consumable library has already begun in [`fabric/pkg/config`](https://github.com/hyperledger/fabric/tree/master/pkg/config) 
+and tracked in [FAB-17628](https://jira.hyperledger.org/browse/FAB-17628). Example usage has been included in the 
+[godocs](https://godoc.org/github.com/hyperledger/fabric/pkg/config#pkg-examples) for this package and can be referenced for external tooling.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+API documentation already exists in the godocs for this package and can be referenced [here](https://godoc.org/github.com/hyperledger/fabric/pkg/config).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Currently many consumers already have a stable method for updating configuration transactions and may be
+unwilling to make amendments to their processes by consuming this library. It could be argued that even if this library
+exists, no external users already using `configtxgen` would want to switch to consuming it. 
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+It is under the impression that eventually it will become too onerous to maintain current workarounds for channel configuration management
+as new features are added around channel config. Rather than having to manually update bits and pieces of workaround
+code, consumption of this library will become a consistent method for adopting new features.
+
+# Prior art
+[prior-art]: #prior-art
+
+All good golang dependencies are built up as a consumable, consistent, well-documented with examples, libraries with effective
```

Comment:
This sentence was difficult for me to parse at first, probably because it jumps right in and mentions 'dependencies', while this library has not explicitly been called a 'dependency' above.